hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -136,15 +139,16 @@ def compat_kw(*args, **kw):
             ),
         )
     elif dtype.kind == "c":
-        # If anyone wants to add a `width` argument to `complex_numbers()`, we would
-        # accept a pull request and add passthrough support for magnitude bounds,
-        # but it's a low priority otherwise.
-        kws = compat_kw("allow_nan", "allow_infinity", "allow_subnormal")
-        if dtype.itemsize == 8:
-            float32 = st.floats(width=32, **kws)
-            result = st.builds(complex, float32, float32)
-        else:
-            result = st.complex_numbers(**kws)
+        result = st.complex_numbers(
+            width=8 * dtype.itemsize,  # convert from bytes to bits
```

Comment:
Sould we do this instead, much like in the float case above?

```suggestion
            width=min(8 * dtype.itemsize, 128),  # convert from bytes to bits
```

Because right now, [a test](https://github.com/HypothesisWorks/hypothesis/actions/runs/4091754172/jobs/7056042667#step:6:927) fail for dtype `"G"`, which is an alias for `"complex256"`.

Reply:
So yes ^^https://github.com/HypothesisWorks/hypothesis/pull/3570#issuecomment-1416830748